### ISSUE TITLE
3x speed regression in RNN predict() and fit() performance in tfjs v0.11.5

### ISSUE THREAD

**Issue by caisq** [tensorflow/tfjs#438]
Compared to @tensorflow/tfjs-node 0.11.4, 0.11.5 shows a dramatic speed regression.

The same bad performance is seen in the latest version (0.11.6).

This can be seen by running the addition-rnn example on different versions of @tensorflow/tfjs. On my machine:

```none
version  -core version  -layers version  Speed & behavior
0.11.6    0.11.6             0.6.6                 getting slower and slower  (IN the end, <200 examples/s)
0.11.5    0.11.5             0.6.5                 behaves similarly to 0.11.6 (<200 examples/s)
0.11.4    0.11.4             0.6.4                 ~850 examples/s
```

So, it appears that somewhere between tfjs 0.11.4 and 0.11.5, there is a certain commit that led to this RNN speed regression. The commit may either be in tfjs-core or tfjs-layers. We should investigate.

**Comment by caisq** [comment]
cc @nsthorat @dsmilkov

**Comment by caisq** [comment]
I haven't dug thoroughly yet, but here are two possible culprits I can find that fit the results of profiling against different versions above.

1. https://github.com/tensorflow/tfjs-core/pull/1080
2. https://github.com/tensorflow/tfjs-layers/pull/217

I think 1 is more likely than 2, because the speed regression affects both predict() and fit() and 2 is relevant to only fit().

**Comment by nsthorat** [comment]
Do you know if there is a memory leak? #1 shouldn't have an impact unless it introduced a leak (however our numerous leak tests pass).

**Comment by caisq** [comment]
Good question. I just tested it at HEAD (which shows the slowdown). See the for-demo PR at:
https://github.com/tensorflow/tfjs-layers/pull/242

There is no memory leak during the fit() call to a model consisting of an LSTM layer.

**Comment by caisq** [comment]
Just verified that this is fixed in @tensorflowjs/tfjs v0.11.7.
